cabal-install/Distribution/Client/FileMonitor.hs

```diff
@@ -199,7 +199,7 @@ monitorFileHashedSearchPath notFoundAtPaths foundAtPath =
 -- files to be monitored (index by their path), and a list of
 -- globs, which monitor may files at once.
 data MonitorStateFileSet
-   = MonitorStateFileSet !(Map FilePath MonitorStateFile)
+   = MonitorStateFileSet ![MonitorStateFile]
```

Comment:
I doubt it'd gain us anything. It's a cost to maintain and unless we have lots of duplicates doesn't save anything.